src/server/middlewares/redirects.ts

```diff
@@ -33,3 +34,32 @@ export const startPageRedirect: IMiddleware<object> = async (ctx) => {
 
   ctx.redirect(`/se${queryStringMaybe}`)
 }
+
+export const manualRedirects: IMiddleware<State, any> = async (ctx, next) => {
+  const permanentRedirects = await getDatasourceEntries('permanent-redirects')
+  const temporaryRedirects = await getDatasourceEntries('temporary-redirects')
+
+  const permanentRedirect = permanentRedirects.datasource_entries.find(
+    (redirect) => redirect.name === ctx.path,
+  )
+
+  if (permanentRedirect) {
+    const queryStringMaybe = ctx.querystring ? '?' + ctx.querystring : ''
+    ctx.redirect(`${permanentRedirect.value}${queryStringMaybe}`)
+    ctx.status = 301
+    return
+  }
+
+  const temporaryRedirect = temporaryRedirects.datasource_entries.find(
+    (redirect) => redirect.name === ctx.path,
+  )
+
+  if (temporaryRedirect) {
+    const queryStringMaybe = ctx.querystring ? '?' + ctx.querystring : ''
+    ctx.redirect(`${temporaryRedirect.value}${queryStringMaybe}`)
+    ctx.status = 302
+    return
+  }
```

Comment:
We do have some duplicate code here, maybe we could generify it a bit just to not repeat ourselves?

Reply:
ofc! 🙌